pkg/controller/clusterdeployment/clusterinstalls.go

```diff
@@ -55,7 +55,9 @@ func (r *ReconcileClusterDeployment) reconcileExistingInstallingClusterInstall(c
 	if met := ci.Spec.ClusterMetadata; met != nil &&
 		met.InfraID != "" &&
 		met.ClusterID != "" &&
+		met.AdminKubeconfigSecretRef != (corev1.LocalObjectReference{}) &&
```

Comment:
This _shouldn't_ be necessary. If the value of the field isn't a pointer, it's the nil value, which is an "empty" struct, but that empty struct implicitly has all its fields (including `Name` in this case) with nil values (in this case `""`) if unset. So TL;DR the next line should be sufficient for the AdminKubeConfigSecretRef field.